tck/methods/account.go

```diff
@@ -0,0 +1,279 @@
+	InitialBalance int64  `json:"initialBalance"`
+}
+type UpdateAccountKeyParams struct {
+	AccountId     string `json:"accountId"`
+	NewPublicKey  string `json:"newPublicKey"`
+	OldPrivateKey string `json:"oldPrivateKey"`
+	NewPrivateKey string `json:"newPrivateKey"`
+}
+type UpdateAccountMemoParams struct {
+	AccountId string `json:"accountId"`
+	Key       string `json:"key"`
+	Memo      string `json:"memo"`
+}
+type DeleteAccountParams struct {
+	AccountId   string `json:"accountId"`
+	AccountKey  string `json:"accountKey"`
+	RecipientId string `json:"recipientId"`
+}
+
+// TODO: Move this struct in dedicated file for hepler structs, or something like this. We want to use it for all methods
+type ErrorStatus struct {
+	Status string `json:"status"`
+}
+
+// ---- Structs to hold different response structures expected from jRPC client ----
+type AccountResponse struct {
+	AccountId string `json:"accountId"`
+	Status    string `json:"status"`
+}
+type AccountInfoResponse struct {
+	AccountID                     string
+	Balance                       string
+	Key                           string
+	AccountMemo                   string
+	MaxAutomaticTokenAssociations uint32
+	AutoRenewPeriod               time.Duration
+}
+
+// Variable to be set to `SetGrpcDeadline` for all transactions
+var threeSec = time.Second * 3
+
+// SetSdkService We set object, which is holding our client params. Pass it by referance, because TCK is dynamically updating it
+func (a *AccountService) SetSdkService(service *SDKService) {
+	a.sdkService = service
+}
+
+// TODO Find better way to handle error. We want some wrapper, which would intercept the error and contruct the correct
+// TODO jrpc error based on that  "jrpc2.Errorf(-32603, "Internal error"/"Hedera error", ErrorStatus{Status: err.Error()})"
+// CreateAccount jRPC method for createAccount
+func (a *AccountService) CreateAccount(_ context.Context, accountParams CreateAccountParams) (*AccountResponse, error) {
+	transaction := hedera.NewAccountCreateTransaction().SetGrpcDeadline(&threeSec)
+
+	if accountParams.PublicKey != "" {
+		key, err := hedera.PublicKeyFromString(accountParams.PublicKey)
+		if err != nil {
+			return nil, jrpc2.Errorf(-32603, "Internal error", ErrorStatus{Status: err.Error()})
+		}
+		transaction.SetKey(key)
+	}
+	if accountParams.InitialBalance != 0 {
+		transaction.SetInitialBalance(hedera.HbarFromTinybar(accountParams.InitialBalance))
+	}
+	if accountParams.ReceiverSignatureRequired {
+		transaction.SetReceiverSignatureRequired(accountParams.ReceiverSignatureRequired)
+	}
+	if accountParams.MaxAutomaticTokenAssociations != 0 {
+		transaction.SetMaxAutomaticTokenAssociations(accountParams.MaxAutomaticTokenAssociations)
+	}
+	if accountParams.StakedAccountId != "" {
+		accountId, err := hedera.AccountIDFromString(accountParams.StakedAccountId)
+		if err != nil {
+			return nil, jrpc2.Errorf(-32603, "Internal error", ErrorStatus{Status: err.Error()})
```

Comment:
If the JSON RPCs have a standard error code like -32603 would define it somewhere. Will improve readability.

Reply:
We can find multiple occurances of this.

Reply:
Done